### PR TITLE
Capri memory error

### LINKED ISSUE #584: correct memory error in caprieval
Currently, a tedious error can occur in the caprieval module when multiple capri jobs are executed in parallel.

### PR DESCRIPTION
**Pull request by mgiulini**
You are about to submit a new Pull Request. Before continuing make sure you read the [contributing guidelines](CONTRIBUTING.md) and you comply with the following criteria:

- [x] You have stick to Python. Talk with us before adding other programming languages to HADDOCK3
- [ ] Your PR is about CNS
- [x] Your code is well documented: proper docstrings and explanatory comments for those tricky parts
- [x] You structured the code into small functions as much as possible. You can use classes if there's a (state) purpose
- [x] code follows our coding style
- [x] You wrote tests for the new code
- [x] `tox` tests pass. *Run `tox` command inside the repository folder*
- [ ] `-test.cfg` examples execute without errors. *Inside `examples/` run `python run_tests.py -b`*
- [x] PR does not add any *install dependencies* unless permission granted by the HADDOCK team
- [x] PR does not break licensing
- [ ] Your PR is about writing documentation for already existing code :fire:
- [x] Your PR is about writing tests for already existing code :godmode:

Closes #584 

The memory error was due to the parallel execution of multiple capri jobs. Apparently, calling the `get_intermolecular_contacts` function can lead to crashes in CAPRI jobs. Additionally, it was found that the same function is sometimes not 100% precise in retrieving the correct contacts.

This PR addresses this problems by:

1. creating a haddock3-specific, memory-friendly `load_contacts` function https://github.com/haddocking/haddock3/blob/f58ccb1d1238ba8c88ee3ad308be463a150b27a2/src/haddock/modules/analysis/caprieval/capri.py#L27-L75 : this function iterates over all the atoms of a reference chain, retrieving the contacts that the atom forms with the target chain. It is slower than the original implementation, but more accurate and less memory-demanding (a contact matrix is not needed anymore)
2. giving the possibility to retrieve the full set of heavy atoms of a protein or DNA molecule (useful for calculating contacts) https://github.com/haddocking/haddock3/blob/f58ccb1d1238ba8c88ee3ad308be463a150b27a2/src/haddock/libs/libalign.py#L65-L106
3. introducing an additional test for caprieval, where two heterogeneous complexes are compared. The values of irmsd and fnat obtained with the previous routines were slightly wrong, while the current ones have been compared to those of another software

### CODE REVIEW

File: src/haddock/libs/libalign.py [hunk 1]
```diff
@@ -61,6 +61,30 @@
 DNA_RES = ["DA", "DC", "DT", "DG"]
```
Comment (amjjbonvin) [inline src/haddock/libs/libalign.py:61]: Do you also have RNA defined?

**Review by amjjbonvin — APPROVED** [review]

File: src/haddock/libs/libalign.py (hunk 1 above)
Comment (mgiulini) [reply src/haddock/libs/libalign.py:61]: > Do you also have RNA defined?

not explicitly, but the function should use by default all the heavy atoms whenever the structure is not dna or protein

Comment (mgiulini) [reply src/haddock/libs/libalign.py:61]: worth to test it by the way!

**Comment by amjjbonvin** [comment]
DNA or RNA is almost the same - I would make it then nucleix acid and add the rna bases to it

=================================
 Prof. A.M.J.J. Bonvin | Faculty of Science
 Scientific Director Bijvoet Centre
 Utrecht University | bonvinlab.org 
 ***@***.*** | T: +31 30 2533859 
=================================

**Review by rvhonorato — APPROVED** [review]
looks good to me

**Pull request merged**